sensor/common/sensor/central_communication_impl.go

```diff
@@ -270,11 +274,23 @@ func (s *centralCommunicationImpl) initialDeduperSync(stream central.SensorServi
 	}
 	log.Info("Waiting for deduper state from Central")
 
-	msg, err := stream.Recv()
+	done := make(chan struct{})
+	var err error
+	var msg *central.MsgToSensor
+	go func() {
+		msg, err = stream.Recv()
+		close(done)
+	}()
+	select {
+	case <-time.After(s.syncTimeout):
+		return errTimeoutWaitingForDeduperState
+	case <-done:
```

Comment:
Would be so nice if gRPC stream offered channels as interface. We could write this as `case msg, more := <-stream.Recv()`

(this is a non-actionable comment, feel free to resolve when you need to merge)

Reply:
Yeah, it would be way prettier than having to have the `done` channel 😞